src/interface/execute-automation-hook/api-execute-automation-hook.ts

```diff
@@ -0,0 +1,187 @@
+import { Stack } from "aws-cdk-lib";
+import { AutomationStepSimulation, AwsApiStep, DataTypeEnum, IAwsInvoker, IExecuteAutomationHook, isBoolean, ISleepHook, isMapList, isNumber, isString, isStringList, isStringMap, ResponseCode, SleepStep } from "../..";
+import { ExecuteAutomationProps, ExecuteAutomationOutputs } from "../execute-automation-hook";
+
+export interface ApiExecuteAutomationProps {
+    readonly awsInvoker: IAwsInvoker;
+    readonly sleepHook: ISleepHook;
+}
+
+interface AutomationExecution {
+    AutomationExecutionStatus: ExecutionStatus;
+    Outputs: Record<string, string[]>;
+}
+
+type ExecutionStatus = 
+"Pending"|
+"InProgress"|
+"Waiting"|
+"Success"|
+"TimedOut"|
+"Cancelling"|
+"Cancelled"|
+"Failed"|
+"PendingApproval"|
+"Approved"|
+"Rejected"|
+"Scheduled"|
+"RunbookInProgress"|
+"PendingChangeCalendarOverride"|
+"ChangeCalendarOverrideApproved"|
+"ChangeCalendarOverrideRejected"|
+"CompletedWithSuccess"|
+"CompletedWithFailure";
+
+/**
+ * ExecuteAutomation implementation using AWS API
+ */
+export class ApiExecuteAutomationHook implements IExecuteAutomationHook {
+    private static readonly RunningStatuses: ExecutionStatus[] = ['Pending', 'InProgress', 'Cancelling', 'Waiting'];
```

Comment:
If you run `npm run eslint` it should fix the spacing (It uses 2 spaces rather than 4)

Reply:
Ran eslint